dbt_semantic_interfaces/parsing/where_filter_parser.py

```diff
@@ -38,10 +50,19 @@ def parse_call_parameter_sets(where_sql_template: str) -> FilterCallParameterSet
 
         def _dimension_call(dimension_name: str, entity_path: Sequence[str] = ()) -> str:
             """Gets called by Jinja when rendering {{ dimension(...) }}."""
+            group_by_item_name = DunderedNameFormatter.parse_name(dimension_name)
+            if len(group_by_item_name.entity_links) != 1:
+                raise ParseWhereFilterException(
+                    WhereFilterParser._exception_message_for_incorrect_format(dimension_name)
+                )
+
             dimension_call_parameter_sets.append(
                 DimensionCallParameterSet(
-                    dimension_reference=DimensionReference(element_name=dimension_name),
-                    entity_path=tuple(EntityReference(element_name=arg) for arg in entity_path),
+                    dimension_reference=DimensionReference(element_name=group_by_item_name.element_name),
+                    entity_path=(
+                        tuple(EntityReference(element_name=arg) for arg in entity_path)
+                        + group_by_item_name.entity_links
```

Comment:
So the primary entity will be the last one in the entity_path tuple. Cool Cool 👍